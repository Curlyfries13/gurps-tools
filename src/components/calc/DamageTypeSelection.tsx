import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import { setDamageType } from 'src/redux/actions/damageActions';
import { RootState } from 'src/redux/reducers';

import { DamageTypes } from 'src/types';

const DamageTypeSelection = ({ damageType, setDamageType }: Props) => {
  const handleTypeChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    setDamageType(event.target.value);
  };

  return (
    <select
      onChange={handleTypeChange}
      value={damageType}
      className='form-select'
      aria-label='damage type selection'
    >
      {Object.keys(DamageTypes).map((key) => {
        return (
          <option value={key} key={key}>
            {key}
          </option>
        );
      })}
    </select>
  );
};

function mapStateToProps(state: RootState) {
  return {
    damageType: state.damageType,
  };
}

const mapDispatchToProps = {
  setDamageType,
};

const connector = connect(mapStateToProps, mapDispatchToProps);
type Props = ConnectedProps<typeof connector>;

export default connector(DamageTypeSelection);
