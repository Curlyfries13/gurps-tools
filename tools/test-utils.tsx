// Using code from https://redux.js.org/recipes/writing-tests#connected-components
import React from 'react';

// test-utils.js
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
// Import your own reducer
//
// console.log(React.createElement);

import * as reducer from 'src/redux/reducers';


const customRender = (
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) => {
  const Wrapper =({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
