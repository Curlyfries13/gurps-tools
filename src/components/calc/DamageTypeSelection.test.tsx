import React from 'react';
import { RenderResult } from '@testing-library/react';
import { render } from 'tools/test-utils';

import DamageTypeSelection from './DamageTypeSelection';

it('matches snapshot', async () => {
  const component: RenderResult = render(<DamageTypeSelection />);
  expect(component).toMatchSnapshot();
});
