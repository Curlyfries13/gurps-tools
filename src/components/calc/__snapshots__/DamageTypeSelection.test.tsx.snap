// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <select
        aria-label="damage type selection"
        class="form-select"
      >
        <option
          value="cr"
        >
          cr
        </option>
        <option
          value="cut"
        >
          cut
        </option>
        <option
          value="imp"
        >
          imp
        </option>
        <option
          value="pi"
        >
          pi
        </option>
        <option
          value="pi+"
        >
          pi+
        </option>
        <option
          value="pi++"
        >
          pi++
        </option>
        <option
          value="pi-"
        >
          pi-
        </option>
        <option
          value="burn"
        >
          burn
        </option>
        <option
          value="cor"
        >
          cor
        </option>
        <option
          value="fat"
        >
          fat
        </option>
        <option
          value="tox"
        >
          tox
        </option>
      </select>
    </div>
  </body>,
  "container": <div>
    <select
      aria-label="damage type selection"
      class="form-select"
    >
      <option
        value="cr"
      >
        cr
      </option>
      <option
        value="cut"
      >
        cut
      </option>
      <option
        value="imp"
      >
        imp
      </option>
      <option
        value="pi"
      >
        pi
      </option>
      <option
        value="pi+"
      >
        pi+
      </option>
      <option
        value="pi++"
      >
        pi++
      </option>
      <option
        value="pi-"
      >
        pi-
      </option>
      <option
        value="burn"
      >
        burn
      </option>
      <option
        value="cor"
      >
        cor
      </option>
      <option
        value="fat"
      >
        fat
      </option>
      <option
        value="tox"
      >
        tox
      </option>
    </select>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
